openwisp_monitoring/device/api/filters.py

```diff
@@ -25,3 +27,23 @@ class MonitoringDeviceFilter(OrganizationManagedFilter):
     class Meta(OrganizationManagedFilter.Meta):
         model = Device
         fields = OrganizationManagedFilter.Meta.fields + ['monitoring__status']
+
+
+class MonitoringNearbyDeviceFilter(OrganizationManagedFilter):
+    distance__lte = filters.NumberFilter(
+        label=_('Distance is less than or equal to'),
+        field_name='distance',
+        lookup_expr='lte',
+    )
+    model = filters.CharFilter(method='filter_model')
+
+    class Meta(OrganizationManagedFilter.Meta):
+        model = Device
+        fields = OrganizationManagedFilter.Meta.fields + [
+            'monitoring__status',
+            'model',
+        ]
+
+    def filter_model(self, queryset, name, value):
+        values = value.split(',')
```

Comment:
Model name could contain a command (`,`). We should use other character for this. I suggest using `+`.